src/components/TimeCounter.tsx

```diff
@@ -1,79 +1,67 @@
-import { StackNavigationProp } from '@react-navigation/stack';
-import dayjs from 'dayjs';
-import React, { useState, useEffect } from 'react';
-import { StyleSheet, Text, View, KeyboardAvoidingView } from 'react-native';
-import { Overlay } from 'react-native-elements';
+import React, { Component } from 'react';
+import { View, StyleSheet } from 'react-native';
 
-import { MainCardStackNavigatorParams, Route } from 'app/consts';
-import { useInterval } from 'app/helpers/useInterval';
-import { typography } from 'app/styles';
+import { palette, fonts } from 'app/styles';
 
-import { Button } from './Button';
-
-const i18n = require('../../loc');
+import { Text } from './Text';
 
 interface Props {
-  timestamp: number;
-  isVisible: boolean;
-  navigation: StackNavigationProp<MainCardStackNavigatorParams, Route.CurrentPin>;
-  onCountFinish?: () => void;
+  value: number;
 }
 
-export const TimeCounter = ({ timestamp, onCountFinish, isVisible, navigation }: Props) => {
-  const currentTimestamp = dayjs().unix();
-  const secondsToCount = (timestamp - currentTimestamp).toFixed(0);
-  const [seconds, setSeconds] = useState(parseInt(secondsToCount));
-
-  const isTimeUp = (): boolean => seconds === 0;
-
-  useInterval(
-    () => {
-      setSeconds(seconds - 1);
-    },
-    seconds > 0 ? 1000 : null,
-  );
-
-  useEffect(() => {
-    if (isTimeUp()) {
-      onFinish();
-    }
-  });
-
-  const onFinish = () => {
-    if (onCountFinish) {
-      onCountFinish();
-    }
-  };
+export class TimeCounter extends Component<Props> {
+  render() {
+    const { value } = this.props;
+    const minutes = Math.floor(value / 60);
+    const seconds = value - minutes * 60;
+    const middleElement = 2;
+    const getStringifiedWithZero = (number: number) => (number < 10 ? `0${number}` : `${number}`);
```

Comment:
`const getStringifiedWithZero = (number: number) => number.toString().padStart(2, '0')`
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/padStart

Note: you don't need this function with dayjs usage described in the comment above